File: core/data/src/main/java/org/mifos/mobilewallet/core/data/fineract/entity/UserEntity.java
package org.mifos.mobilewallet.core.data.fineract.entity;

import java.util.ArrayList;
import java.util.List;

public class UserEntity {

    private long userId;
    private boolean authenticated;
    private String username;
    private String base64EncodedAuthenticationKey;
    private List<String> permissions = new ArrayList<String>();

    public String getUserName() {
        return username;
    }

    public long getUserId() {
        return userId;
    }

    public boolean isAuthenticated() {
        return authenticated;
    }

    public String getBase64EncodedAuthenticationKey() {
        return base64EncodedAuthenticationKey;
    }


    public List<String> getPermissions() {
        return permissions;
    }

    @Override
    public String toString() {
        return "User{" +
                "username='" + username + '\'' +
                ", userId=" + userId +
                ", base64EncodedAuthenticationKey='" + base64EncodedAuthenticationKey + '\'' +
                ", authenticated=" + authenticated +
                '}';
    }
}

File: core/src/main/java/org/mifos/mobilewallet/core/data/fineract/entity/UserEntity.kt
package org.mifos.mobilewallet.core.data.fineract.entity

import java.util.*

data class UserEntity(
        val userId: Long? = null,
        val isAuthenticated: Boolean? = null,
        val userName: String? = null,
        val base64EncodedAuthenticationKey: String? = null,
        val permissions: List<String> = ArrayList())